lib/ImageWorker.js

```diff
@@ -79,8 +79,8 @@ class ImageWorker extends Component<ImageWorkerProps, ImageWorkerState> {
     const image = new Image();
     this.image = image;
     
-    image.decode !== undefined ? image.decode().then(this.onLoad) : image.onload = this.onLoad;
     image.src = url;
+    image.decode !== undefined ? image.decode().then(this.onLoad).catch(this.onLoad) : image.onload = this.onLoad;
```

Comment:
!== undefined isnt needed. Just image.decode will suffice

Reply:
The explicit check gives higher performance, see here: 
![c7oofqixqaariog](https://user-images.githubusercontent.com/502146/36075422-29f1c0f4-0f4f-11e8-9726-b603a0a2c274.jpg)

Reply:
Really.. interesting.. thanks. Learnt something new today..

Reply:
I dint know about this. Nice one